msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/HttpResponse.java

```diff
@@ -62,11 +62,12 @@ List<String> getHeader(String key) {
         return headers.get(key);
     }
 
-    JSONObject getBodyAsJson() {
-        try {
-            return JSONObjectUtils.parse(this.body());
-        } catch (ParseException e) {
-            throw new RuntimeException(e);
+    Map<String, String> getBodyAsMap() {
+        try (JsonReader reader = JsonProviders.createReader(this.body)) {
+            reader.nextToken();
+            return reader.readMap(JsonReader::getString);
+        } catch (IOException e) {
+            throw new MsalClientException("Could not parse JSON from HttpResponse body: " + e.getMessage(), AuthenticationErrorCode.INVALID_JSON);
```

Comment:
Looks like a good candidate for a helper method since there's an almost identical copy in `TestHelper.java`?